cpg_infra/billing_aggregator/aggregate/hail.py

```diff
@@ -190,6 +195,25 @@ def from_request(request: Request):
     return asyncio.new_event_loop().run_until_complete(main(start, end))
 
 
+async def process_batch_ids(batch_ids: list[str]):
+    """
+    Process batch ids
+    """
+    # locate start and end time as main will need them
+    start = None
+    end = None
+    for batch_id in batch_ids:
+        batch = await utils.get_batch_by_id(batch_id, token=utils.get_hail_token())
+        start_time = utils.parse_hail_time(batch['time_created'])
+        end_time = utils.parse_hail_time(batch['time_completed'])
+        if start is None or start_time < start:
+            start = start_time
+        if end is None or end_time > end:
+            end = end_time
+
+    return await main(start, end, batch_ids=batch_ids)
```

Comment:
The start / end are not used batch_ids flow of `main` - so I don't think there's a need for this.
And if for some reason you do, set them in the `main` function where you get the batches.

Reply:
Yes, we need the start and end date to retrieve 'stored_ids' from BQ table, prior to calling each batch separately. This was intruded in this PR to optimise those multiple BQ calls with each Hail Batch individually.